src/main/java/org/apache/bcel/classfile/RecordComponentInfo.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.bcel.classfile;
+
+import java.io.DataInput;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+import org.apache.bcel.Const;
+
+/**
+ * Record component info from a record. Instances from this class maps
+ * every component from a given record.
+ *
+ * @see <a href="https://docs.oracle.com/javase/specs/jvms/se14/preview/specs/records-jvms.html#jvms-4.7.30">
+ *      The Java Virtual Machine Specification, Java SE 14 Edition, Records (preview)</a>
+ */
+public class RecordComponentInfo implements Node {
+
+    private final int index;
+    private final int descriptorIndex;
+    private final Attribute[] attributes;
+    private final ConstantPool constantPool;
+
+    /**
+     * Constructs object from input stream.
+     *
+     * @param input        Input stream
+     * @param constantPool Array of constants
+     * @throws IOException if an I/O error occurs.
+     */
+    public RecordComponentInfo(final DataInput input, ConstantPool constantPool) throws IOException {
+        this.index = input.readUnsignedShort();
+        this.descriptorIndex = input.readUnsignedShort();
+        final int attributesCount = input.readUnsignedShort();
+        this.attributes = new Attribute[attributesCount];
+        for (int j = 0; j < attributesCount; j++) {
+            attributes[j] = Attribute.readAttribute(input, constantPool);
+        }
+        this.constantPool = constantPool;
+    }
+
+    public void dump(DataOutputStream file) throws IOException {
+        file.writeShort(index);
+        file.writeShort(descriptorIndex);
+        file.writeShort(attributes.length);
+        for (final Attribute attribute : attributes) {
+            attribute.dump(file);
+        }
+    }
+
+    @Override
+    public void accept(Visitor v) {
+        v.visitRecordComponent(this);
+    }
+
+    public int getIndex() {
+        return index;
+    }
+
+    public int getDescriptorIndex() {
```

Comment:
Hi @PabloNicolasDiaz 
This method is neither used or tested, how about removing until it is needed? 
Or, is it used by you in an other app?
It is also missing a Javadoc.